File: mns-core/src/test/java/cz/fb/manaus/core/model/RunnerPricesTest.java
package cz.fb.manaus.core.model;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.Test;

import java.util.Collection;
import java.util.List;

import static org.hamcrest.CoreMatchers.containsString;
import static org.junit.Assert.assertThat;

public class RunnerPricesTest {

    public static final ObjectMapper MAPPER = new ObjectMapper();

    public static RunnerPrices create(long selectionId, Collection<Price> prices, Double matched, Double lastMatchedPrice) {
        var rp = new RunnerPrices();
        rp.setSelectionId(selectionId);
        rp.setPrices(prices);
        rp.setMatchedAmount(matched);
        rp.setLastMatchedPrice(lastMatchedPrice);
        return rp;
    }

    @Test
    public void testMarshallJson() throws Exception {
        RunnerPrices prices = create(111, List.of(), 100d, 5d);
        String json = MAPPER.writeValueAsString(prices);
        assertThat(json, containsString("111"));
    }
}

File: mns-core/src/test/java/cz/fb/manaus/core/model/RunnerPricesTest.kt
package cz.fb.manaus.core.model

import org.junit.jupiter.api.Test
import kotlin.test.assertEquals


class RunnerPricesTest {

    @Test
    fun homogeneous() {
        val back = runnerPrices.first().by(Side.BACK)
        assertEquals(2, back.prices.size)
        assertEquals(Side.BACK, back.prices.first().side)
    }

    @Test
    fun `best price`() {
        val lay = runnerPrices.first().by(Side.LAY)
        val back = runnerPrices.first().by(Side.BACK)
        assertEquals(2.5, back.bestPrice?.price)
        assertEquals(3.5, lay.bestPrice?.price)
    }
}